crates/ruff_linter/src/rules/refurb/rules/slice_to_remove_prefix_or_suffix.rs

```diff
@@ -0,0 +1,479 @@
+        }),
+        (
+            AffixKind::StartsWith,
+            ast::Expr::Call(ast::ExprCall {
+                range: _,
+                func,
+                arguments,
+            }),
+            _,
+        ) => {
+            semantic.match_builtin_expr(func, "len")
+                && arguments.len() == 1
+                && arguments.find_positional(0).is_some_and(|arg| {
+                    let compr_affix = ast::comparable::ComparableExpr::from(affix);
+                    let compr_arg = ast::comparable::ComparableExpr::from(arg);
+                    compr_affix == compr_arg
+                })
+        }
+        (
+            AffixKind::EndsWith,
+            ast::Expr::UnaryOp(ast::ExprUnaryOp {
+                op: ast::UnaryOp::USub,
+                operand,
+                range: _,
+            }),
+            ast::Expr::StringLiteral(ast::ExprStringLiteral {
+                range: _,
+                value: string_val,
+            }),
+        ) => operand.as_number_literal_expr().is_some_and(
+            |ast::ExprNumberLiteral { value, .. }| {
+                value.as_int().is_some_and(|x| {
+                    // Only support prefix removal for size at most `u32::MAX`
+                    u32::try_from(string_val.len()).is_ok_and(|length| x == &ast::Int::from(length))
+                })
+            },
+        ),
+        (
+            AffixKind::EndsWith,
+            ast::Expr::UnaryOp(ast::ExprUnaryOp {
+                op: ast::UnaryOp::USub,
+                operand,
+                range: _,
+            }),
+            _,
+        ) => operand.as_call_expr().is_some_and(
+            |ast::ExprCall {
+                 range: _,
+                 func,
+                 arguments,
+             }| {
+                semantic.match_builtin_expr(func, "len")
+                    && arguments.len() == 1
+                    && arguments.find_positional(0).is_some_and(|arg| {
+                        let compr_affix = ast::comparable::ComparableExpr::from(affix);
+                        let compr_arg = ast::comparable::ComparableExpr::from(arg);
+                        compr_affix == compr_arg
+                    })
+            },
+        ),
+        _ => false,
+    }
+}
+
+/// Generates the source code string
+/// ```python
+/// text = text.removeprefix(prefix)
+/// ```
+/// or
+/// ```python
+/// text = text.removesuffix(prefix)
+/// ```
+/// as appropriate.
+fn generate_assignment_with_removeaffix(
+    text: &ast::Expr,
+    affix_query: &AffixQuery,
+    locator: &Locator,
+) -> String {
+    let text_str = locator.slice(text);
+    let affix_str = locator.slice(affix_query.affix);
+
+    match affix_query.kind {
+        AffixKind::StartsWith => format!("{text_str} = {text_str}.removeprefix({affix_str})"),
+        AffixKind::EndsWith => format!("{text_str} = {text_str}.removesuffix({affix_str})"),
+    }
```

Comment:
We can do this slightly more simply now that you've added the `AffixKind::replacement()` method!

```suggestion
    let replacement = affix_query.kind.replacement();
    format!("{text_str} = {text_str}.{replacement}({affix_str})")
```